src/xclim/indices/run_length.py

```diff
@@ -1660,12 +1660,13 @@ def index_of_date(
     """
     if date is None:
         return np.array([default])
-    try:
+    if len(date.split("-")) == 2:
+        date = f"2000-{date}"
```

Comment:
"2000"? Is this solely to get a valid datetime string?

Reply:
Yep. Now that I think of it, we should use a leap year so we can parse `02-29`...

```suggestion
        date = f"1840-{date}"
```
In honor of the Act of Union adopted by the British Parliament in July 1840.